libs/datamodel/parser-database/src/walkers/index_name.rs

```diff
@@ -0,0 +1,64 @@
+use super::ScalarFieldWalker;
+
+/// Represents the name of an index
+
+#[derive(Debug, Hash, PartialEq, Eq)]
+pub enum IndexName<'ast> {
+    /// Explicit index name defined in the datamodel
+    Explicit(&'ast str),
+    /// Generated index name based on the fields of the index.
+    /// Only used on compound ids
```

Comment:
Comment needs to be update to: "A `Generated` index name can only contain a name if it's compound"